lib/core/src/chain_swap.rs

```diff
@@ -781,9 +781,10 @@ impl ChainSwapHandler {
     }
 
     // Updates the swap without state transition validation
-    pub(crate) fn update_swap(&self, updated_swap: ChainSwap) -> Result<(), PaymentError> {
+    pub(crate) fn update_swap(&self, updated_swap: ChainSwap) -> Result<bool, PaymentError> {
```

Comment:
Instead of changing the update_swaps functions  did you consider only trigger the update_wallet_info on the event of PendingPayment ?

Reply:
I can test this. I think it would be good to cover all payment related events